spotbugs/src/main/java/edu/umd/cs/findbugs/ba/XFactory.java

```diff
@@ -223,6 +227,10 @@ private boolean isCalledDirectlyOrIndirectly(@CheckForNull ClassDescriptor clazz
 
     }
 
+    /**
+     * @deprecated This method does not work as expected. Will be removed in 5.x release.
```

Comment:
According to IntelliJ IDEA, we have no code depending on this method.
But it's a public method, so I'll keep this method at this moment, and will remove it in the next major update.